src/nebula-hooks/virtualized-table/use-page-info.ts

```diff
@@ -0,0 +1,28 @@
+import { useMemo, useState } from '@nebula.js/stardust';
+import { MAX_PAGE_SIZE } from '../../table/virtualized-table/constants';
+import { PageInfo, TableLayout } from '../../types';
+
+const usePageInfo = (layout: TableLayout, shouldRender: boolean) => {
+  const [page, setPage] = useState(0);
+  const pageSize = Math.min(MAX_PAGE_SIZE, layout.qHyperCube.qSize.qcy);
+  const pageInfo = useMemo<PageInfo>(
+    () => ({
+      page,
+      rowsPerPage: pageSize,
+      rowsPerPageOptions: [],
+    }),
+    [pageSize, page]
+  );
+
+  if (shouldRender) {
+    // Guard against new layout that contains fewer pages then previous layout
+    const lastPage = Math.floor(layout.qHyperCube.qSize.qcy / MAX_PAGE_SIZE);
```

Comment:
I think this still should be `Math.ceil()`. because lets say you have 240 rows and you are showing 100 per page.

you should have 3 pages for that but with Math.floor() -> you will only get 2 pages and never being able to see the last 40 rows. because of line 20 check below 👇 

```js
// lastPage -> 2
// currentPage -> 2
if(2 < 2) {
   // never will execute this if check, but we still have last 40 rows left to show in page 3
}

```

Reply:
I was thinking so at first to. BUT the first page is actually `0`. So when you do `Math.ceil(240 / 100)` you get `3`. Which points to a page that does not exist.

The users however will still page that starts on `1`.